catalyst/contrib/dl/callbacks/cmc_callback.py

```diff
@@ -0,0 +1,180 @@
+from typing import List
+
+import torch
+
+from catalyst.core import IRunner
+from catalyst.core.callback import CallbackOrder
+from catalyst.dl import Callback
+from catalyst.utils.metrics.cmc_score import cmc_score
+
+
+class CMCScoreCallback(Callback):
+    """
+    Cumulative Matching Characteristics callback
+
+    You should use it with `ControlFlowCallback`
+    and add all query/gallery sets to loaders.
+    Loaders should contain "is_query" and "label" key.
+
+    .. code-block:: python
+
+        runner.train(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            loaders=loaders,
+            logdir="./logdir",
+            num_epochs=5,
+            callbacks=[
+                ControlFlowCallback(
+                    base_callback=CMCScoreCallback(
+                        topk_args=[1, 5]
+                    ),
+                    loaders=["split_1", "split_2"]
+                )
+            ]
+        )
+    """
+
+    def __init__(
+        self,
+        embeddings_key: str = "embeddings",
+        labels_key: str = "labels",
+        is_query_key: str = "query",
+        prefix: str = "cmc",
+        topk_args: List[int] = None,
+    ):
+        """
+        This callback was designed to count
+        cumulative matching characteristics.
+        If current object is from query your dataset
+        should output `True` in `is_query_key`
+        and false if current object is from gallery.
```

Comment:
Probably it will be better to create abstract dataset with the following behabiour?